common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -368,6 +373,55 @@ impl NavigationController {
     }
 }
 
+/// Functionality for the navigation controller that is exported. (or will be exported)
+impl NavigationRecording {
```

Comment:
Minor organizational nit, but I usually prefer to have any `impl`s right after the `struct` definition.

I think we can also *probably* move both of these into a `recording` module under `navigation_controller` at this point; there's enough code now, and `NavigationRecording` isn't really a model.